src/overflow_check_conditional.rs

```diff
@@ -0,0 +1,41 @@
+use rustc::lint::*;
+use syntax::ast::*;
+use utils::{span_lint, span_help_and_lint, snippet, span_lint_and_then};
+
+/// **What it does:** This lint finds classic overflow checks.
+///
+/// **Why is this bad?** Most classic overflow checks would cause the compiler to panic.
+///
+/// **Known problems:** None.
+///
+/// **Example:** `a + b < a`
+declare_lint!(pub OVERFLOW_CHECK_CONDTIONAL, Warn,
+              "Using overflow checks which are likely to panic");
+
+#[derive(Copy, Clone)]
+pub struct OverflowCheckConditional;
+
+impl LintPass for OverflowCheckConditional {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(OVERFLOW_CHECK_CONDTIONAL)
+    }
+}
+
+impl LateLintPass for OverflowCheckConditional {
+    fn check_expr(&mut self, cx: &LateContext, expr: &Expr) {
+    	if_let_chain! {[
+    		let ExprKind::Binary(ref op, ref first, ref second) = expr.node,
+    		let BinOpKind::Lt = op.node,
+    		let ExprKind::Binary(ref op2, ref add1, ref add2) = first.node,
+    		let BinOpKind::Add = op2.node,
+    		let ExprKind::Path(_,ref path1) = add1.node,
+    		let ExprKind::Path(_, ref path2) = add2.node,
+    		let ExprKind::Path(_, ref path3) = second.node
+    	], {
+    		if ((&path1.segments[0]).identifier == (&path3.segments[0]).identifier) ||
+    		((&path2.segments[0]).identifier == (&path3.segments[0]).identifier) {
+        		// Some lint
```

Comment:
You must check the type of the operands here (search for `expr_ty` in Clippy) and lint only for integers.